### PR TITLE
docs: Use rST instead of Markdown for README and ROADMAP

### LINKED ISSUE #818: Bug Report: Sphinx v3.0.0 breaks docs build
# Description

The release of Sphinx [`v3.0.0`](https://github.com/sphinx-doc/sphinx/releases/tag/v3.0.0) [breaks the build of the docs with](https://github.com/scikit-hep/pyhf/runs/562901637?check_suite_focus=true#step:6:11)

```pytb
Exception occurred:
  File "/opt/hostedtoolcache/Python/3.7.6/x64/lib/python3.7/site-packages/sphinx/application.py", line 1069, in add_source_parser
    self.registry.add_source_parser(*args, **kwargs)
TypeError: add_source_parser() takes 2 positional arguments but 3 were given
The full traceback has been saved in /tmp/sphinx-err-gu866l6p.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
```

which when run locally the generated log is

```
$ cat /tmp/sphinx-err-blah.log
# Sphinx version: 3.0.0
# Python version: 3.7.5 (CPython)
# Docutils version: 0.16 release
# Jinja2 version: 2.11.1
# Last messages:

# Loaded extensions:
Traceback (most recent call last):
  File "/home/feickert/.venvs/pyhf-dev/lib/python3.7/site-packages/sphinx/cmd/build.py", line 279, in build_main
    args.tags, args.verbosity, args.jobs, args.keep_going)
  File "/home/feickert/.venvs/pyhf-dev/lib/python3.7/site-packages/sphinx/application.py", line 244, in __init__
    self.setup_extension(extension)
  File "/home/feickert/.venvs/pyhf-dev/lib/python3.7/site-packages/sphinx/application.py", line 398, in setup_extension
    self.registry.load_extension(self, extname)
  File "/home/feickert/.venvs/pyhf-dev/lib/python3.7/site-packages/sphinx/registry.py", line 414, in load_extension
    metadata = setup(app)
  File "/home/feickert/.venvs/pyhf-dev/lib/python3.7/site-packages/m2r.py", line 652, in setup
    app.add_source_parser('.md', M2RParser)
  File "/home/feickert/.venvs/pyhf-dev/lib/python3.7/site-packages/sphinx/application.py", line 1069, in add_source_parser
    self.registry.add_source_parser(*args, **kwargs)
TypeError: add_source_parser() takes 2 positional arguments but 3 were given
```

# Steps to Reproduce

Try to build the docs with Sphinx `v3.0.0` or check the nightly CI builds.

# Checklist

- [x] Run `git fetch` to get the most up to date version of `master`
- [x] Searched through existing Issues to confirm this is not a duplicate issue
- [x] Filled out the Description, Expected Behavior, Actual Behavior, and Steps to Reproduce sections above or have edited/removed them in a way that fully describes the issue

### PR DESCRIPTION
**Pull request by kratsg**
# Description

Resolves #818. We drop the `m2r` dependency. This fixes the sphinx builds.

Docs build at: https://pyhf.readthedocs.io/en/docs-newsphinx/

# Checklist Before Requesting Reviewer

- [x] Tests are passing
- [x] "WIP" removed from the title of the pull request
- [x] Selected an Assignee for the PR to be responsible for the log summary

# Before Merging

For the PR Assignees:

- [x] Summarize commit messages into a comprehensive review of the PR

```
* Write README and ROADMAP in rST instead of Markdown
* Drop m2r dependency
* Avoid unused arguments error from pyflakes (new in pyflakes v2.2.0)
* Run apt-get clean on Linux runners due to GitHub Actions disk space regression
   - c.f. https://github.community/t5/GitHub-Actions/GitHub-Actions-Failing-with-Errno-28-No-space-left-on-device/m-p/54516
```

### CODE REVIEW

**Comment by matthewfeickert** [comment]
Seems we have [another docs issue](https://github.com/scikit-hep/pyhf/pull/819/checks?check_run_id=564956890#step:6:111)

```
/home/runner/work/pyhf/pyhf/docs/conf.py:29: RemovedInSphinx40Warning: The app.add_stylesheet() is deprecated. Please use app.add_css_file() instead.
  'https://cdnjs.cloudflare.com/ajax/libs/github-fork-ribbon-css/0.2.2/gh-fork-ribbon.min.css'
```

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+   nobs = 55, b = 50, db = 7, nom_sig = 10.
+
+.. image:: https://raw.githubusercontent.com/scikit-hep/pyhf/master/docs/_static/img/manual_1bin_55_50_7.png
+   :alt: manual
+   :width: 40 %
+
+.. image:: https://raw.githubusercontent.com/scikit-hep/pyhf/master/docs/_static/img/hfh_1bin_55_50_7.png
+   :alt: manual
+   :width: 40 %
+   :align: right
```
Comment (matthewfeickert) [inline README.rst:102]:
```suggestion
.. image:: https://raw.githubusercontent.com/scikit-hep/pyhf/master/docs/_static/img/manual_1bin_55_50_7.png
   :alt: manual
   :width: 70 %
   :align: center

.. image:: https://raw.githubusercontent.com/scikit-hep/pyhf/master/docs/_static/img/hfh_1bin_55_50_7.png
   :alt: manual
   :width: 70 %
   :align: center
```

Comment (matthewfeickert) [reply README.rst:102]: I think that 70% across two lines is closer to what we currently have. The centering is a deviation from the current, but I like the look better.

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+   bin 2: nobs = 145, b = 150, db = 20., nom_sig = 45.
+
+.. image:: https://raw.githubusercontent.com/scikit-hep/pyhf/master/docs/_static/img/manual_2_bin_100.0_145.0_100.0_150.0_15.0_20.0_30.0_45.0.png
+   :alt: manual
+   :width: 40 %
+
+.. image:: https://raw.githubusercontent.com/scikit-hep/pyhf/master/docs/_static/img/hfh_2_bin_100.0_145.0_100.0_150.0_15.0_20.0_30.0_45.0.png
+   :alt: manual
+   :width: 40 %
+   :align: right
```
Comment (matthewfeickert) [inline README.rst:119]:
```suggestion
.. image:: https://raw.githubusercontent.com/scikit-hep/pyhf/master/docs/_static/img/manual_2_bin_100.0_145.0_100.0_150.0_15.0_20.0_30.0_45.0.png
   :alt: manual
   :width: 70 %
   :align: center

.. image:: https://raw.githubusercontent.com/scikit-hep/pyhf/master/docs/_static/img/hfh_2_bin_100.0_145.0_100.0_150.0_15.0_20.0_30.0_45.0.png
   :alt: manual
   :width: 70 %
   :align: center
```

I think that 70% across two lines is closer to what we currently have. The centering is a deviation from the current, but I like the look better.

File: docs/governance/ROADMAP.rst [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+      components <https://github.com/iris-hep/project-milestones/issues/17>`__”
+      [2020-Q3 → 2020-Q4]
+
+Roadmap as Gantt Chart
+~~~~~~~~~~~~~~~~~~~~~~
+
+.. figure:: https://user-images.githubusercontent.com/5142394/64583069-53049180-d355-11e9-8b39-8b2a4599e21e.png
+   :alt: pyhf_AS_gantt
+
+   pyhf_AS_gantt
```
Comment (matthewfeickert) [inline docs/governance/ROADMAP.rst:145]:
```suggestion
```
Dangling remnant?

File: README.rst [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+  - [x] Import from XML + ROOT via `uproot <https://github.com/scikit-hep/uproot>`__
+  - [x] ShapeFactor
+  - [x] StatError
+  - [x] Lumi Uncertainty
+
+Computational Backends:
+  - [x] NumPy
+  - [x] PyTorch
+  - [x] TensorFlow
+  - [x] JAX
```
Comment (matthewfeickert) [inline README.rst:64]:
```suggestion
  - ☒ HistoSys
  - ☒ OverallSys
  - ☒ ShapeSys
  - ☒ NormFactor
  - ☒ Multiple Channels
  - ☒ Import from XML + ROOT via `uproot <https://github.com/scikit-hep/uproot>`__
  - ☒ ShapeFactor
  - ☒ StatError
  - ☒ Lumi Uncertainty

Computational Backends:
  - ☒ NumPy
  - ☒ PyTorch
  - ☒ TensorFlow
  - ☒ JAX
```

For consistency with the rest of the docs maybe switch from [x] to ☒?

**Comment by matthewfeickert** [comment]
Before this goes in, it might be worth checking how [Black is able to use Makrdown](https://github.com/psf/black/blob/master/setup.py) but doesn't use `m2r`.

**Comment by matthewfeickert** [comment]
@lukasheinrich Given @kratsg's [thoughts here](https://github.com/scikit-hep/pyhf/issues/818#issuecomment-611027473) and my inability to find a solution over the last 2 days what are your thoughts on this PR?

**Comment by kanishk16** [comment]
Just a suggestion if we switch to rST, I believe it would be nice if we could possibly have checkboxes just as we had in case of markdown (ref. [3rd last comment on this issue at Stack Overflow](https://stackoverflow.com/questions/6904552/how-to-create-a-checklist-in-restructuredtext-rest))...I have tested for README.rst and unsure about how it renders on Github but the docs get rendered out as below with some minor inconsistencies of space gaps (as displayed in the Todo img.) that could be very well be customized or ignored if that's not an issue.  

```
What does it support
--------------------

.. |check| raw:: html 
    
    <input checked="" disabled="" type="checkbox"/>

Implemented variations:
  |check| HistoSys

  |check| OverallSys
  
  |check| ShapeSys
  
  |check| NormFactor
  
  |check| Multiple Channels
  
  |check| Import from XML + ROOT via `uproot <https://github.com/scikit-hep/uproot>`__
  
  |check| ShapeFactor
  
  |check| StatError
  
  |check| Lumi Uncertainty

Computational Backends:
  |check| NumPy

  |check| PyTorch
  
  |check| TensorFlow
  
  |check| JAX
```
image

```
Todo
----

.. |uncheck| raw:: html

    <input disabled="" type="checkbox"/>

|uncheck| StatConfig

|uncheck| Non-asymptotic calculators

```

image

**Comment by matthewfeickert** [comment]
@kanishk16 I think that looks really great! Nice find.

@kratsg @lukasheinrich While I don't think any of us are 100% happy with this solution at the moment, it does work and pass the tests, so I think that we should just adopt rST for now and move on while checking to see if there might be ways to correct the image sizes on GitHub or bring back Markdown in the future. Any objections?

**Comment by matthewfeickert** [comment]
It looks like using ``.. raw:: html` like we did for the Markdown will work just fine (unless I'm missing something). I need to revert the HTML check boxes though, as they don't get rendered by GitHub.

**Comment by matthewfeickert** [comment]
>  I need to revert the HTML check boxes though, as they don't get rendered by GitHub.

Actually, if we're fine seeing lists like this on GitHub:

Implemented variations:
- HistoSys
- OverallSys
- ShapeSys
- NormFactor

then we can keep the HTML check mark boxes, and they'll just render on the docs and PyPI. Sound good?

**Comment by matthewfeickert** [comment]
I'll also post this on my GitHub Community Forum Issue, but the issue with the GitHub disk space regression was only for the Linux runners, and if additional space is manually cleaned up on them then things run fine again. If I remove `/swapfile` and run `apt-get clean` `df -h` shows

```
Filesystem      Size  Used Avail Use% Mounted on
udev            3.4G     0  3.4G   0% /dev
tmpfs           695M  960K  694M   1% /run
/dev/sda1        84G   64G   20G  77% /
tmpfs           3.4G  8.0K  3.4G   1% /dev/shm
tmpfs           5.0M     0  5.0M   0% /run/lock
tmpfs           3.4G     0  3.4G   0% /sys/fs/cgroup
/dev/loop1       40M   40M     0 100% /snap/hub/43
/dev/loop0       94M   94M     0 100% /snap/core/8935
/dev/sda15      105M  3.6M  101M   4% /boot/efi
/dev/sdb1        14G   41M   13G   1% /mnt
```

and if I only run `apt-get clean`

```
Filesystem      Size  Used Avail Use% Mounted on
udev            3.4G     0  3.4G   0% /dev
tmpfs           695M  960K  694M   1% /run
/dev/sda1        84G   72G   12G  87% /
tmpfs           3.4G  8.0K  3.4G   1% /dev/shm
tmpfs           5.0M     0  5.0M   0% /run/lock
tmpfs           3.4G     0  3.4G   0% /sys/fs/cgroup
/dev/loop0       40M   40M     0 100% /snap/hub/43
/dev/loop1       94M   94M     0 100% /snap/core/8935
/dev/sda15      105M  3.6M  101M   4% /boot/efi
/dev/sdb1        14G   41M   13G   1% /mnt
```

which appears to be enough space for the install to run fine. This is an annoying hack and GitHub needs to fix this disk space bug, but it works for the time being.

File: docs/governance/ROADMAP.rst [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+   (November 4-8th, 2019)
+-  |check| `Poster at CHEP
+   2019 <https://indico.cern.ch/event/773049/contributions/3476180/>`__
+   (November 4-8th, 2019)
+
+.. |Seibert Heirarchy of Needs SciPy 2019| image:: https://pbs.twimg.com/media/EBYojw8XUAERJhZ?format=png
+
+.. |check| raw:: html
+
+    <input checked=""  type="checkbox">
```
Comment (kratsg) [inline docs/governance/ROADMAP.rst:165]:
```suggestion
    <input checked=""  type="checkbox" disabled="true">
```

File: docs/governance/ROADMAP.rst [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+
+.. |uncheck| raw:: html
+
+    <input type="checkbox">
```
Comment (kratsg) [inline docs/governance/ROADMAP.rst:169]:
```suggestion
    <input type="checkbox" disabled="true">
```

**Comment by matthewfeickert** [comment]
> Shouldn't we be adding `disabled="true"` to `README.rst` as well or am I just missing something?

We should. I missed this earlier.

**Pull request merged**